pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -520,6 +529,10 @@ func (gce *GCECloud) EnsureLoadBalancer(name, region string, requestedIP net.IP,
 	// Thus, we have to tear down the forwarding rule if either it or the target
 	// pool needs to be updated.
 	if fwdRuleExists && (fwdRuleNeedsUpdate || tpNeedsUpdate) {
+		// If we have to delete the forwarding rule, and something should fail
+		// before we recreate it, don't release the IP.  That way we can come
+		// back to it later.
+		orphanStaticIP = true
```

Comment:
I know this is getting a little far out there, but I still think this won't work in all cases. Consider:
1. We determine we need to update the forwarding rule
2. We create a static IP to reserve the forwarding rule's IP address across its deletion and recreation, since we can't modify it in place
3. We delete the forwarding rule
4. We fail to recreate the forwarding rule, so we error out of the function, leaving the static IP in place (yay!)
5. The service controller waits a few seconds, then retries
6. The call above to gce.createOrPromoteStaticIP returns the IP that got orphaned last time, as desired
7. This conditional never triggers, because we deleted the forwarding rule last time, so fwdRuleExists evaluates false
8. We fail to recreate the forwarding rule, but don't leave the IP around because we never set orphanStaticIP to true